src/tasks/submit/flashbots.rs

```diff
@@ -117,38 +117,63 @@ impl FlashbotsTask {
                 debug!("upstream task gone - exiting flashbots task");
                 break;
             };
-            let span = sim_result.span();
-            span_debug!(span, "received sim result");
+
+            let span = sim_result.sim_env.clone_span();
+
+            // Don't submit empty blocks
+            if sim_result.block.is_empty() {
+                counter!("signet.builder.flashbots.empty_block").increment(1);
+                span_debug!(span, "received empty block - skipping");
+                continue;
+            }
+
+            let host_block_number = sim_result.host_block_number();
```

Comment:
these are already on the span in the sim_env, no?

https://github.com/init4tech/builder/blob/main/src/tasks/env.rs#L114